algebra/group/coset.lean

```diff
@@ -0,0 +1,174 @@
+import algebra.group.subgroup
+
+open set
+
+universe u
+variable {G : Type u}
+
+def lcoset [has_mul G] (a : G) (S : set G) : set G := image (λ x, a * x) S
+def rcoset [has_mul G] (S : set G) (a : G) : set G := image (λ x, x * a) S
+
+namespace coset_notation
+
+infix * := lcoset
+infix * := rcoset
+
+end coset_notation
```

Comment:
This doesn't work. You can't namespace notations in lean 3, and this will ruin multiplication everywhere else. You must use local notation, but then they can't be the same symbol. Try `*l` and `*r` or something.

Reply:
Fixed in https://github.com/leanprover/mathlib/pull/94/commits/8d602aa4e8ca01a70c61a99d2107ffb261c8df71 and https://github.com/leanprover/mathlib/pull/94/commits/2b7b65b28040726dd801095589a15ffcaa49229f.